cmd/agent/main.go

```diff
@@ -21,37 +21,56 @@
 package main
 
 import (
-	"flag"
 	"runtime"
 
 	"github.com/uber/jaeger-lib/metrics/go-kit"
 	"github.com/uber/jaeger-lib/metrics/go-kit/expvar"
 	"go.uber.org/zap"
 
 	"github.com/uber/jaeger/cmd/agent/app"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 )
 
-func main() {
-	builder := app.NewBuilder()
-	builder.Bind(flag.CommandLine)
-	flag.Parse()
+var (
+	builder = app.NewBuilder()
+)
+
+func init() {
```

Comment:
I think it's a good practice to configure cobra/viper in init function. To do that I need access to object created by `app.NewBuilder` because its flags depend on it.

Do you want to move everything to main?